src/OrleansRuntime/Streams/QueueBalancer/IStreamQueueBalancerFactory.cs

```diff
@@ -0,0 +1,22 @@
+﻿using Orleans.Providers;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Streams
+{
+    public interface IStreamQueueBalancerFactory
+    {
```

Comment:
still think this should be an interface, instead of a factory delegate... 
I think factory as a delegate will be good if it is part of a bigger class's property, it is light weight that way. But that is not true in this PR. 
User inject a factory instance, instead of a factory delegate fits more into OOP patterns